build/toolchain/gcc_toolchain.gni

```diff
@@ -280,16 +280,20 @@ template("gcc_toolchain") {
 
     # When invoking this toolchain not as the default one, these args will be
     # passed to the build. They are ignored when this is the default toolchain.
-    toolchain_args = {
-      current_cpu = invoker.toolchain_cpu
-      current_os = invoker.toolchain_os
+    if (defined(invoker.toolchain_args)) {
+      toolchain_args = invoker.toolchain_args
```

Comment:
nit: I think instead of letting the invoker specify the toolchain args, can we just let the invoker specify some other object like `extra_toolchain_args` and append them here?

I think GN will let you do that with `+=`, but maybe I'm wrong. It seems strange to let the invoker change the target_os/target_cpu/is_clang this way.

Reply:
Just tried this. Unfortunately you can't `+=` a block. But I think I was able to find a way using `forward_variables_from`